shared/src/packages/users/libs/validation-schemas/user-sign-up.validation-schema.ts

```diff
@@ -1,21 +1,45 @@
 import joi from 'joi';
 
-import { UserValidationMessage } from '../enums/enums.js';
 import { type UserSignUpRequestDto } from '../types/types.js';
+import { Regexp as REGEXP } from './enums/regexp.enum.js';
+import { UserValidationMessage } from './enums/user-validation-message.enum.js';
 
-const userSignUp = joi.object<UserSignUpRequestDto, true>({
+const passwordPattern = new RegExp(REGEXP.PASSWORD);
+const namePattern = new RegExp(REGEXP.NAME);
+const phonePattern = new RegExp(REGEXP.PHONE);
+
+const userSignUpRules = {
+  phone: joi.string().trim().required().pattern(phonePattern).messages({
+    'string.empty': UserValidationMessage.REQUIRED,
+    'string.pattern.base': UserValidationMessage.PHONE_NOT_VALID,
+  }),
+  password: joi.string().trim().required().pattern(passwordPattern).messages({
+    'string.empty': UserValidationMessage.REQUIRED,
+    'string.pattern.base': UserValidationMessage.PASSWORD_NOT_VALID,
+  }),
   email: joi
     .string()
     .trim()
+    .min(5)
+    .max(254)
     .email({ tlds: { allow: false } })
-    .required(),
-  phone: joi.string().trim().required().messages({
-    'string.empty': UserValidationMessage.PHONE_REQUIRED,
+    .required()
+    .messages({
+      'string.empty': UserValidationMessage.REQUIRED,
+      'string.email': UserValidationMessage.EMAIL_NOT_VALID,
+      'string.min': UserValidationMessage.EMAIL_NOT_VALID,
+      'string.max': UserValidationMessage.EMAIL_NOT_VALID,
+    }),
+  firstName: joi.string().trim().required().pattern(namePattern).messages({
+    'string.empty': UserValidationMessage.REQUIRED,
+    'string.pattern.base': UserValidationMessage.NAME_NOT_VALID,
   }),
-  password: joi.string().trim().required(),
-  firstName: joi.string().trim().required(),
-  lastName: joi.string().trim().required(),
-  groupId: joi.number().required(),
-});
+  lastName: joi.string().trim().required().pattern(namePattern).messages({
+    'string.empty': UserValidationMessage.REQUIRED,
+    'string.pattern.base': UserValidationMessage.NAME_NOT_VALID,
+  }),
+};
+
+const userSignUp = joi.object<UserSignUpRequestDto, true>(userSignUpRules);
 
-export { userSignUp };
+export { userSignUp, userSignUpRules };
```

Comment:
Do you need to export signUp rules?

Reply:
I need it as rules shared between signup customer and business, as an option I can move them to separate file

Reply:
moved common rules to separate file